libraries/Microsoft.Bot.Expressions/BuiltInFunctions.cs

```diff
@@ -1338,10 +1330,6 @@ private static (object value, string error) Substring(Expression expression, IMe
                         }
                     }
                 }
-                else
-                {
-                    error = $"{expression.Children[0]} is not a string.";
```

Comment:
>error = $"{expression.Children[0]} is not a string."; [](start = 20, length = 53)

Why did you drop this?  At runtime you can up with a non-string.

Reply:
This was checking to see if the object returned was not a string.  The TryEvaluate<string>() will coerce the result to a string, aka, the number 31 => "31".  

---
In reply to: [367282705](https://github.com/microsoft/botbuilder-dotnet/pull/3253#discussion_r367282705) [](ancestors = 367282705)